compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -418,6 +418,11 @@ void addMultiTilingExpertPassPipeline(OpPassManager &funcPassManager,
             createLLVMCPUSplitReductionPass(clEnableReassociateFpReductions));
         funcPassManager.addPass(createLLVMCPUTileRootAndFuseInputOperandsPass(
             static_cast<IREE::CPU::TilingLevel>(i)));
+        // Tile all the reduction ops for target vector sizes. It is a nop for
+        // rootOp because it is already tiled with the same tile sizes. It
+        // ensures that all the dimensions are tiled in all the reduction ops.
+        funcPassManager.addPass(
+            createLLVMCPUTilePass(static_cast<IREE::CPU::TilingLevel>(i)));
```

Comment:
It is not NOP if the dimension size is not divisible by tile size, because it becomes dynamic shapes and the for loops can't be folded away.

I think we need to provide an option to `LLVMCPUTilePass` that allows only tiling on non-root ops.